### PR TITLE
Fixed #28965 -- Update cookie date format to follow RFC 7231.

### PR DESCRIPTION
**Pull request by ogonkov**
https://code.djangoproject.com/ticket/28965

### CODE REVIEW

File: django/utils/http.py [hunk 1]
```diff
@@ -110,16 +110,18 @@ def urlencode(query, doseq=False):
 
 def cookie_date(epoch_seconds=None):
     """
-    Format the time to ensure compatibility with Netscape's cookie standard.
+    Format the time
```
Comment (timgraham) [inline django/utils/http.py]: add: "according to...."

This function is also documented at `docs/ref/utils.txt`. The documentation should be updated, including a ``.. versionchanged:: 2.1` annotation. A release note should be added in `docs/releases/2.1.txt` under backwards-incompatible changes.

File: django/utils/http.py [hunk 2]
```diff
@@ -110,16 +110,18 @@ def urlencode(query, doseq=False):
...
     `epoch_seconds` is a floating point number expressed in seconds since the
     epoch, in UTC - such as that outputted by time.time(). If set to None, it
     defaults to the current time.
 
-    Output a string in the format 'Wdy, DD-Mon-YYYY HH:MM:SS GMT'.
+    Output a string in the format 'Wdy, DD Mon YYYY HH:MM:SS GMT'.
     """
-    rfcdate = formatdate(epoch_seconds)
-    return '%s-%s-%s GMT' % (rfcdate[:7], rfcdate[8:11], rfcdate[12:25])
+    return formatdate(
```
Comment (timgraham) [inline django/utils/http.py]: Single line looks fine here. For consistency with other code, I'd pass`epoch_seconds` as an arg.

Comment (ogonkov) [reply django/utils/http.py]: You mean like this?
```py
formatdate(epoch_seconds, usegmt=True)
```

Comment (timgraham) [reply django/utils/http.py]: Yes

File: django/utils/http.py (hunk 1 above)
Comment (ogonkov) [reply django/utils/http.py]: Looks like `cookie_date` is same to `http_date`, and should be replaced by it. Haven't noticed it before.

Comment (ogonkov) [reply django/utils/http.py]: I need mark it as deprecated? I think i couldn't just remove it

Comment (timgraham) [reply django/utils/http.py]: Deprecation could be a separate ticket/task. For now, just update the documentation to indicate how it has changed.

**Comment by timgraham** [comment]
I updated the patch to switch usages of `cookie_date()` to `http_date()` and added a commit to deprecate the now unused `cookie_date()` function. Please review if you'd like to.

**Pull request merged**